hikari/impl/entity_factory.py

```diff
@@ -1843,49 +1963,62 @@ def deserialize_command_interaction(
-                for user_id, member_payload in raw_members.items():
-                    assert guild_id is not None
-                    user_id = snowflakes.Snowflake(user_id)
-                    members[user_id] = self._deserialize_interaction_member(
-                        member_payload, user=users[user_id], guild_id=guild_id
-                    )
+        return command_interactions.CommandInteraction(
+            app=self._app,
+            application_id=snowflakes.Snowflake(payload["application_id"]),
+            id=snowflakes.Snowflake(payload["id"]),
+            type=base_interactions.InteractionType(payload["type"]),
+            guild_id=guild_id,
+            guild_locale=payload.get("guild_locale", "en-US"),
+            locale=payload["locale"],
+            channel_id=snowflakes.Snowflake(payload["channel_id"]),
+            member=member,
+            user=user,
+            token=payload["token"],
+            version=payload["version"],
+            command_id=snowflakes.Snowflake(data_payload["id"]),
+            command_name=data_payload["name"],
+            command_type=commands.CommandType(data_payload.get("type", commands.CommandType.SLASH)),
+            options=options,
+            resolved=resolved,
+            target_id=target_id,
+        )
 
-            if raw_roles := resolved_payload.get("roles"):
-                assert guild_id is not None
-                roles_iter = (self.deserialize_role(role, guild_id=guild_id) for role in raw_roles.values())
-                roles = {r.id: r for r in roles_iter}
+    def deserialize_autocomplete_interaction(
+        self, payload: data_binding.JSONObject
+    ) -> command_interactions.AutocompleteInteraction:
+        data_payload = payload["data"]
 
-            else:
-                roles = {}
+        guild_id: typing.Optional[snowflakes.Snowflake] = None
+        if raw_guild_id := payload.get("guild_id"):
+            guild_id = snowflakes.Snowflake(raw_guild_id)
 
-            resolved = command_interactions.ResolvedOptionData(
-                channels=channels,
-                members=members,
-                users=users,
-                roles=roles,
-            )
+        options: typing.Optional[typing.List[command_interactions.AutocompleteInteractionOption]] = None
+        if raw_options := data_payload.get("options"):
+            options = [self._deserialize_autocomplete_interaction_option(option) for option in raw_options]
 
-        return command_interactions.CommandInteraction(
+        member: typing.Optional[base_interactions.InteractionMember]
+        if member_payload := payload.get("member"):
+            assert guild_id is not None
+            member = self._deserialize_interaction_member(member_payload, guild_id=guild_id)
+            # See https://github.com/discord/discord-api-docs/pull/2568
+            user = member.user
+
+        else:
+            member = None
+            user = self.deserialize_user(payload["user"])
```

Comment:
Missing a test case for when autocomplete interactions have a user instead of a member (pretty much just a copy paste from existing tests job for this one tbf)